src/Packetery/Module/Framework/WcSessionTrait.php

```diff
@@ -4,14 +4,34 @@
 
 namespace Packetery\Module\Framework;
 
+use WC_Session;
+use WC_Session_Handler;
+
 trait WcSessionTrait {
 	/**
-	 * @param string $key
-	 * @param mixed  $value
-	 *
-	 * @return void
+	 * @return WC_Session|WC_Session_Handler|null
 	 */
+	public function session() {
+		return WC()->session;
+	}
+
+	/**
+	 * @return array|string
+	 */
+	public function sessionGet( string $key ) {
+		return WC()->session->get( $key );
+	}
+
 	public function sessionSet( string $key, $value ): void {
 		WC()->session->set( $key, $value );
 	}
+
+	/**
+	 * Phpdoc is not reliable.
+	 *
+	 * @return int|string
+	 */
+	public function sessionGetCustomerId() {
+		return WC()->session->get_customer_id();
```

Comment:
ta session může být null nevím kdy a proč ale nebylo by to vhodné ošetřit?

Reply:
volá se tam před použitím initializeSession. chtěl jsem udržet adaptery bez logiky, ale asi by se dalo inicializovat v rámci session metody a tu pak všude použít